src/components/editMeasure/EditMeasure.tsx

```diff
@@ -27,15 +28,29 @@ export default function EditMeasure() {
   const [measure, setMeasure] = useState<Measure>(null);
   const [loading, setLoading] = useState<boolean>(true);
 
+  let history = useHistory();
+  const [notFound, setNotFound] = useState<boolean>(false);
+
   useEffect(() => {
     if (!measure) {
-      measureServiceApi.fetchMeasure(id).then((value: Measure) => {
-        setMeasure(value);
-        setLoading(false);
-      });
+      measureServiceApi
+        .fetchMeasure(id)
+        .then((value: Measure) => {
+          setMeasure(value);
+          setLoading(false);
+        })
+        .catch((err) => {
+          if (err.toString().includes("404")) {
+            setNotFound(true);
+          }
+        });
     }
   }, [measureServiceApi, id, measure]);
 
+  if (notFound) {
+    history.push("/404");
+  }
+
```

Comment:
Need to be covered in unit test ?

Reply:
@RohitKandimalla The new test case is working, thanks so much for your kind help!!